### DISCUSSION TITLE
How do you make windows 100% width + height with python?

### DISCUSSION THREAD

**Discussion by godalming123** [wmww/gtk-layer-shell#134] [Q&A] [answered]
How do you make windows 100% width + height with python? Also is there any documentation?

**Comment by Consolatis** [comment, marked as answer]
Could be used to implement a desktop background with icons and context menu:

```python3
#!/usr/bin/env python3

import gi

gi.require_version("Gtk", "3.0")
gi.require_version('GtkLayerShell', '0.1')
from gi.repository import Gtk, GtkLayerShell

if __name__ == '__main__':
	win = Gtk.Window()
	GtkLayerShell.init_for_window(win)
	for anchor in ('LEFT', 'RIGHT', 'TOP', 'BOTTOM'):
		GtkLayerShell.set_anchor(win, getattr(GtkLayerShell.Edge, anchor), True)
	GtkLayerShell.set_layer(win, GtkLayerShell.Layer.BOTTOM)
	win.show_all()
	Gtk.main()
```

Regarding documentation: There is http://lazka.github.io/pgi-docs/#GtkLayerShell-0.1

**Comment by wmww** [comment]
Consolatis's answer looks good to me. Unfortunately we do not have documentation for each language. The C documentation is in the header file [here](https://github.com/wmww/gtk-layer-shell/blob/master/include/gtk-layer-shell.h), which should be pretty easy to translate to Python. You might also want to look at the [layer shell protocol](https://github.com/wmww/gtk-layer-shell/blob/master/protocol/wlr-layer-shell-unstable-v1.xml) for more details. Finally, the [Python example](https://github.com/wmww/gtk-layer-shell/blob/master/examples/simple-example.py) is a good place to start.

**Reply by godalming123** [reply]
Thanks
